### PR TITLE
Ec2 tagger

### PR DESCRIPTION
**Pull request by mattpep**
Utility to give a name to unnamed EC2 instances to make identifying them in the AWS console easier. The name is calculated from the environment and the application tags.

### CODE REVIEW

File: ec2-tools/tagger.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import sys
+import boto.ec2
+import os.path
+from optparse import OptionParser
+import ConfigParser
+
+
+def get_ec2_connection(account):
+    config = ConfigParser.ConfigParser()
+    credentials = os.path.expanduser("~") + "/.aws/credentials"
```
Comment (SteveMarshall) [inline ec2-tools/tagger.py]: The should probably use `os.path.join` instead of hard-coded `/`es?

File: ec2-tools/tagger.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        print str
+        text = sys.stdin.readline().strip().lower()
+        if text == 'yes' or text == 'y':
+            return True
+        elif text == 'no' or text == 'n':
+            return False
+
+
+def calculate_name_for_instance(tags):
+    """
```
Comment (SteveMarshall) [inline ec2-tools/tagger.py]: You have a blank line here, but not for `prompt()`; is that a PEP8-ism, or unintentional?

File: ec2-tools/tagger.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        The application name is picked from the first value found from the tags
+        Apps, Application, App.
+
+        Returns None if no sensible guesses could be made
+    """
+    app_tags = ['Apps', 'Application', 'App']
+    found_app_tags = [s for s in app_tags if s in tags]
+    if len(found_app_tags) > 0 and 'Env' in tags:
+        return "{}-{}".format(tags[found_app_tags[0]], tags['Env'])
```
Comment (SteveMarshall) [inline ec2-tools/tagger.py]: Might be nice if this used named substitutions so your intent is more explicit?

File: ec2-tools/tagger.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        print "\nERROR: Please specify an AWS account (e.g. -a mojdsd)"
+        sys.exit(1)
+
+    con = get_ec2_connection(str(options.aws))
+    instances = con.get_only_instances()
+    for inst in instances:
+        if 'Name' in inst.tags:
+            print "Inst {} already has a name: {}".format(
+                inst.id, inst.tags['Name'])
+        else:
```
Comment (SteveMarshall) [inline ec2-tools/tagger.py]: I might like to see the entire `else` leg pulled out to a separate function, something like `set_instance_name(instance)` (although that name makes it feel a bit like there should be a corresponding getter…).

File: ec2-tools/tagger.py (hunk 1 above)
Comment (ltsampros) [inline ec2-tools/tagger.py]: I think that specifying a credentials file is not needed as boto does this for you already, see http://boto.cloudhackers.com/en/latest/boto_config_tut.html#introduction

I think all you need to use a different account is to use the correct env vars.

File: ec2-tools/tagger.py (hunk 2 above)
Comment (mattpep) [inline ec2-tools/tagger.py]: It's not specified either way (AFAICT) in PEP8. But I've resolved the inconsistency.

File: ec2-tools/tagger.py (hunk 4 above)
Comment (mattpep) [inline ec2-tools/tagger.py]: I've gone with that approach, despite the lack of symmetry re getters/setters. The method was getting a bit long.

**Comment by SteveMarshall** [comment]
There are still a few unnamed substitutions throughout that it'd be good to make named. As for `set_instance_name`… Maybe that should be `apply_name_to_instance` (removing the setter-ambiguity)?

I'd also like to see the changes in 5d02e28 rebased into their respective commits. (Top tip: [git can auto-squash fix commits](https://robots.thoughtbot.com/autosquashing-git-commits).)

Other than that, looks good to me! 👍

**Comment by mattpep** [comment]
@SteveMarshall ready for another pass.

**Comment by SteveMarshall** [comment]
@mattpep Much, much better! The only things I'd like to see now are consistent use of named substitutions throughout, and commit messages more in line with [the GDS styleguide](https://github.com/alphagov/styleguides/blob/master/git.md).

**Comment by SteveMarshall** [comment]
😎

**Pull request merged**